From c8cca4cd3fe75f4c337c03079b5abcfd4f1e2f2f Mon Sep 17 00:00:00 2001
From: "Andrei V. Lepikhov" <lepihov@gmail.com>
Date: Sat, 11 Apr 2026 16:51:32 +0200
Subject: [PATCH 4/4] Skip path precheck

---
 src/backend/optimizer/util/pathnode.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/backend/optimizer/util/pathnode.c b/src/backend/optimizer/util/pathnode.c
index becc3b973fe..7b30cb26d62 100644
--- a/src/backend/optimizer/util/pathnode.c
+++ b/src/backend/optimizer/util/pathnode.c
@@ -880,6 +880,7 @@ add_path(RelOptInfo *parent_rel, Path *new_path)
  * At the time this is called, we haven't actually built a Path structure,
  * so the required information has to be passed piecemeal.
  */
+#ifndef CHAOS_MODE
 bool
 add_path_precheck(RelOptInfo *parent_rel, int disabled_nodes,
 				  Cost startup_cost, Cost total_cost,
@@ -947,7 +948,7 @@ add_path_precheck(RelOptInfo *parent_rel, int disabled_nodes,
 
 	return true;
 }
-#if 0
+#else
 bool
 add_path_precheck(RelOptInfo *parent_rel, int disabled_nodes,
 				  Cost startup_cost, Cost total_cost,
@@ -1237,6 +1238,7 @@ add_partial_path(RelOptInfo *parent_rel, Path *new_path)
  * sure we don't add a partial path if there's already a complete path that
  * dominates it, since in that case the proposed path is surely a loser.
  */
+#ifndef CHAOS_MODE
 bool
 add_partial_path_precheck(RelOptInfo *parent_rel, int disabled_nodes,
 						  Cost startup_cost, Cost total_cost, List *pathkeys)
@@ -1340,7 +1342,7 @@ add_partial_path_precheck(RelOptInfo *parent_rel, int disabled_nodes,
 
 	return true;
 }
-#if 0
+#else
 bool
 add_partial_path_precheck(RelOptInfo *parent_rel, int disabled_nodes,
 						  Cost startup_cost, Cost total_cost, List *pathkeys)
-- 
2.51.0

